src/persist/src/indexed/columnar/parquet.rs

```diff
@@ -0,0 +1,171 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Apache Parquet encodings and utils for persist data
+
+use std::io::{Read, Seek, Write};
+
+use arrow2::io::parquet::read::RecordReader;
+use arrow2::io::parquet::write::RowGroupIterator;
```

Comment:
Interesting! We're using the `parquet2` to write but `arrow2` to read? Why is that? (no critique, just asking about the reasons behind it, since I don't know enough about `parque2`, and the documentation is a bit sparse)

Reply:
This is a great question. The current state is a little bit of a hodgepodge. I started by doing both entirely though arrow2, but the arrow2 (write_file) method for writing a parquet file embeds the arrow schema in it (which is huge in size and unnecessary for us). To skip doing that, I had to directly call the parquet2 method (also called write_file) that is internally used in the arrow2 one. You can see the arrow2 one here, it really doesn't do anything other than shove metadata we don't want in: https://docs.rs/arrow2/latest/src/arrow2/io/parquet/write/mod.rs.html#65-93

Note that there's two main bits to writing a parquet file: the metadata and the batches. For both read and write, the batches go through arrow2 because it has a bunch of wrappers that do exactly what we'd do anyway: namely `Iterator<Item = Result<RowGroupIter<'a, E>, E>>` (see https://docs.rs/parquet2/latest/parquet2/write/fn.write_file.html).